reporting/v1/service.proto

```diff
@@ -0,0 +1,99 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package org.packetbroker.reporting.v1;
+
+import "google/protobuf/wrappers.proto";
+
+option go_package = "go.packetbroker.org/api/reporting;reportingpb";
+
+message MonthYear {
+  // Calendar month (1 tot 12).
+  uint32 month = 1;
+  // Calendar year.
+  uint32 year = 2;
+}
+
+message Today {}
+
+message Last30Days {}
+
+message MonthPeriod {
+  // From month (inclusive).
+  MonthYear from = 1;
+  // To month (inclusive).
+  MonthYear to = 2;
+}
+
+message GetRoutedMessagesRequest {
+  // LoRa Alliance NetID of the Forwarder Member.
+  // When unset, match any Forwarder.
+  google.protobuf.UInt32Value forwarder_net_id = 1;
+  // Tenant ID managed by the Forwarder Member.
+  // When unset, match any Forwarder tenant.
+  google.protobuf.StringValue forwarder_tenant_id = 2;
+  // LoRa Alliance NetID of the Home Network Member.
+  // When unset, match any Home Network.
+  google.protobuf.UInt32Value home_network_net_id = 3;
+  // Tenant ID managed by the Home Network Member.
+  // When unset, match any Home Network tenant.
+  google.protobuf.StringValue home_network_tenant_id = 4;
+
+  oneof time {
+    Today today = 5;
+    Last30Days last_30_days = 6;
+    MonthPeriod period = 7;
+  }
+}
+
+enum PHYPayloadType {
+  JOIN_REQUEST = 0;
+  JOIN_ACCEPT = 1;
+  MAC_PAYLOAD = 2;
+}
+
+message RoutedMessagesRecord {
+  // Month that this record applies to.
+  // If unset, this record applies to the selected moving window (e.g. today, last 30 days).
+  MonthYear month = 1;
+  // LoRa Alliance NetID of the Forwarder Member.
+  uint32 forwarder_net_id = 2;
+  // Tenant ID managed by the Forwarder Member.
+  string forwarder_tenant_id = 3;
+  // LoRa Alliance NetID of the Home Network Member.
+  uint32 home_network_net_id = 4;
+  // Tenant ID managed by the Home Network Member.
+  string home_network_tenant_id = 5;
+  // PHYPayload type.
+  PHYPayloadType phy_payload_type = 6;
+  // Count of all messages that are routed by Packet Broker.
+  uint64 routed_count = 7;
+  // Count of all messages processed successfully by the target.
+  uint64 processed_success_count = 8;
+  // Count of all messages processed with an error reported by the target.
+  // For uplink messages, the key corresponds to org.packetbroker.v3.UplinkMessageProcessingError.
+  // For downlink messages, the key corresponds to org.packetbroker.v3.DownlinkMessageProcessingError.
+  map<uint32, uint64> processed_error_count = 9;
```

Comment:
Even with the comment, it's difficult to understand what's happening here.
Wouldn't it be more intuitive to have something like this?

```proto
message RoutedMessagesRecord {
	// ...
	
	message UplinkErrorCount {
		org.packetbroker.v3.UplinkMessageProcessingError error_type = 1;
		uint64 count = 2;
	}
	  
	message DownlinkErrorCount {
		org.packetbroker.v3.DownlinkMessageProcessingError error_type = 1;
		uint64 count = 2;
	}

	oneof processed_error_count { // oneof not needed if we add fields for each type (see review comment above)
		repeated UplinkErrorCount processed_uplink_error_count = 9;
		repeated DownlinkErrorCount processed_downlink_error_count = 10;
	}
}
```